lib/src/ruby/api.cc

```diff
@@ -485,21 +486,32 @@ namespace facter { namespace ruby {
         }
         LOG_DEBUG("ruby was found at \"%1%\".", ruby);
 
-        bool success;
-        string output, none;
-        tie(success, output, none) = execute(ruby, { "-e", "print File.join(RbConfig::CONFIG['"+libruby_configdir()+"'], RbConfig::CONFIG['LIBRUBY_SO'])" });
+        bool success, loaded = false;
+        string possible_lib_ruby_dirs, ruby_output, none;
+        vector<string> lib_ruby_dirs = libruby_config_variables();
+
+        // Build the list of directories to query for so we only have to execute ruby once
+        for (auto it = lib_ruby_dirs.begin(); it != lib_ruby_dirs.end(); ++it) {
+            possible_lib_ruby_dirs += "'"+*it+"',";
+        }
+
+        tie(success, ruby_output, none) = execute(ruby, { "-e", "print(["+possible_lib_ruby_dirs+"].find do |name| dir = RbConfig::CONFIG[name]; next unless dir; file = File.join(dir, RbConfig::CONFIG['LIBRUBY_SO']); break file if File.exist? file; false end)" });
         if (!success) {
-            LOG_WARNING("ruby failed to run: %1%", output);
+            LOG_WARNING("ruby failed to run: %1%", ruby_output);
             return library;
         }
 
         boost::system::error_code ec;
-        if (!exists(output, ec) || is_directory(output, ec)) {
```

Comment:
I think this code can be restored and remove the `loaded` flag (perhaps change `ruby_output` back to `output` now).